src/main/java/com/commercetools/sync/inventories/InventorySync.java

```diff
@@ -0,0 +1,444 @@
+     * as described in {@link InventorySync#syncDrafts(List)}.
+     *
+     * @param inventories  {@link List} of {@link InventoryEntry} that you would like to sync into your CTP project.
+     * @see InventorySync#syncDrafts(List)
+     */
+    @Override
+    protected CompletionStage<InventorySyncStatistics> process(@Nonnull final List<InventoryEntry> inventories) {
+        final List<InventoryEntryDraft> drafts = transformToDrafts(inventories);
+        return processDrafts(drafts);
+    }
+
+    /**
+     * Iterates through the whole {@code inventories} list and accumulates its valid drafts to batches. Every batch
+     * is then processed by {@link InventorySync#processBatch(List)}. For invalid drafts from {@code inventories}
+     * "processed" and "failed" counters from statistics are incremented and error callback is executed. Valid draft
+     * is a {@link InventoryEntryDraft} object that is not {@code null} and its SKU is not empty.
+     *
+     * @param inventories {@link List} of {@link InventoryEntryDraft} resources that would be synced into CTP project.
+     * @return {@link CompletionStage} with {@link InventorySyncStatistics} holding statistics of all sync
+     *                                           processes performed by this sync instance
+     */
+    @Nonnull
+    private CompletionStage<InventorySyncStatistics> splitToBatchesAndProcess(@Nonnull final List<InventoryEntryDraft>
+                                                                                      inventories) {
+        List<InventoryEntryDraft> accumulator = new ArrayList<>(syncOptions.getBatchSize());
+        final List<CompletableFuture<Void>> completableFutures = new ArrayList<>();
+        for (InventoryEntryDraft entry : inventories) {
+            if (entry != null) {
+                if (isNotEmpty(entry.getSku())) {
+                    accumulator.add(entry);
+                    if (accumulator.size() == syncOptions.getBatchSize()) {
+                        completableFutures.add(processBatch(accumulator).toCompletableFuture());
+                        accumulator = new ArrayList<>(syncOptions.getBatchSize());
+                    }
+                } else {
+                    statistics.incrementProcessed();
+                    statistics.incrementFailed();
+                    handleFailure("Failed to process inventory entry without sku", null);
+                }
+            } else {
+                statistics.incrementProcessed();
+                statistics.incrementFailed();
+                handleFailure("Failed to process null object", null);
+            }
+        }
+        if (!accumulator.isEmpty()) {
+            completableFutures.add(processBatch(accumulator).toCompletableFuture());
+        }
+        return CompletableFuture
+            .allOf(completableFutures.toArray(new CompletableFuture[completableFutures.size()]))
```

Comment:
Could you please create issue for that and assign me? I would like to close this PR as it is grows dangerously.